### DISCUSSION TITLE
AuthorInText citations with numeric CSL styles

### DISCUSSION THREAD

**Discussion by mxpiotrowski** [jgm/pandoc#9850] [General]
In contrast to author-date styles, with numeric CSL styles, there’s no difference between `@Klaus1966b` and `[@Klaus1966b]`: both produce just something like “[42]”.  This makes it hard to switch between author-date and numeric styles.

I was under the impression that this was a limitation of CSL, so I wrote [a Lua filter](https://github.com/mxpiotrowski/author-numeric) to scratch this itch.  However, responding to my announcement on Mastodon, Bruce D'Arcus commented that it should work as expected, i.e., `@Klaus1966b` with a numeric style should produce “Klaus [42]” rather than just “[42].”

I see that a similar issue has been brought up before in #8331, although I’d just expect the behavior above rather than the more complex one described in this issue.

Is this possibly a bug in citeproc?

**Comment by jgm** [comment]
Also relevant: https://github.com/jgm/citeproc/issues/131

**Reply by jgm** [reply]
It's been a while since I've thought about this, but I suspect the problem is the following.  citeproc would need to extract the atuhor name.  But how should it be presented?  CSL has a whole host of different settings that affect how author names are formatted (initials or not? et al or not, and and what stage? etc.).  With author-date citation styles the style is already telling you how to format the author names, and we just need to extract this.  With numerical styles we'd have to...make something up, I guess.

I also seem to recall many people wanting this to work like:  "1 says blah blah [2; 3]."

I'd be happy to get @bdarcus  feedback, though.

**Comment by jgm** [comment]
Some more even more ancient references for completeness:

https://github.com/jgm/pandoc-citeproc/issues/209#issuecomment-172368678
https://github.com/jgm/pandoc-citeproc/issues/125
https://github.com/jgm/pandoc-citeproc/issues/133

**Comment by mxpiotrowski** [comment]
Thanks for your reply and the references!

So it seems that I wasn't completely wrong when I wrote the filter, as I couldn't see a way to do it with CSL.  I agree, the “correct” rendering of an in-text numeric citation can’t easily be inferred from the style—it’s likely different from the bibliography.

I guess a filter then actually might be the best solution.  In fact, it should be quite simple to use a template for customizing the rendering.

**Comment by bdarcus** [comment]
Yeah, I didn't think about the "how to render the author?" question. May need to wait for a solution in CSL proper.
